File: backend/src/main/java/com/cabinetmedical/service/SpecialityService.java
package com.cabinetmedical.service;

import com.cabinetmedical.dto.SpecialityDTO;

import java.util.List;

/**
 * Interface du service de gestion des spécialités médicales
 */
public interface SpecialityService {

    /**
     * Récupère toutes les spécialités triées par nom
     *
     * @return Liste de toutes les spécialités
     */
    List<SpecialityDTO> getAllSpecialities();
}

File: frontend/src/services/specialityService.ts
import { Specialite } from '../types';
import apiClient from './apiClient';

/**
 * Récupère la liste de toutes les spécialités médicales
 * @returns Liste des spécialités triées par nom
 */
export const getAllSpecialities = async (): Promise<Specialite[]> => {
  try {
    const response = await apiClient.get<Specialite[]>('/speciality/allSpecialities');
    return response.data;
  } catch (error) {
    console.error('Error fetching specialities:', error);
    throw error;
  }
};

export default {
  getAllSpecialities,
};

